dependencies/CMakeLists.txt

```diff
@@ -319,34 +326,99 @@ else()
                                                --with-colpack=${CMAKE_INSTALL_PREFIX}/colpack
                                                --with-boost=no
             BUILD_COMMAND ${CMAKE_MAKE_PROGRAM} ${BUILD_FLAGS}
-            INSTALL_COMMAND ${CMAKE_MAKE_PROGRAM} install)
+            INSTALL_COMMAND ${CMAKE_MAKE_PROGRAM} install
+            )
     endif()
 
     if(SUPERBUILD_ipopt)
-        # TODO --enable-debug if building in Debug.
-        # TODO must have gfortran for MUMPS (brew install gcc).
-        # TODO CMake documentation says "Whether the current working directory
-        # is preserved between commands is not defined. Behavior of shell
-        # operators like && is not defined."
-        # Patch the scripts that download Metis and MUMPS to use our
-        # Sourceforge mirror. The original links are not reliable.
-        ExternalProject_Add(ipopt
-            URL https://www.coin-or.org/download/source/Ipopt/Ipopt-3.12.8.zip
-            INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/ipopt"
-            PATCH_COMMAND cd <SOURCE_DIR>/ThirdParty/Metis && patch < ${CMAKE_SOURCE_DIR}/get.Metis.patch
-                  COMMAND cd <SOURCE_DIR>/ThirdParty/Metis && ./get.Metis
-                  COMMAND cd <SOURCE_DIR>/ThirdParty/Mumps && patch < ${CMAKE_SOURCE_DIR}/get.Mumps.patch
-                  COMMAND cd <SOURCE_DIR>/ThirdParty/Mumps && ./get.Mumps
-            # GCC 10 generates a warning when compling MUMPS that we must suppress using
-            # -fallow-argument-mismatch.
-            # Suppress warnings treated as errors in Clang/LLVM with -Wno-error=implicit-function-declaration
-            CONFIGURE_COMMAND <SOURCE_DIR>/configure --prefix=<INSTALL_DIR> ADD_FFLAGS=-fallow-argument-mismatch ADD_CFLAGS=-Wno-error=implicit-function-declaration
-            BUILD_COMMAND ${CMAKE_MAKE_PROGRAM} ${BUILD_FLAGS}
-            INSTALL_COMMAND ${CMAKE_MAKE_PROGRAM} install)
+
+        # Check for arm64 architecture
+        execute_process(
+            COMMAND uname -m
+            OUTPUT_VARIABLE MACHINE_ARCH
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+            )
+
+        # For ARM-based Macs, use different version of METIS and MUMPS than
+        # those automatically pulled from IPOPT's ThirdParty submodules.
+        if(APPLE AND MACHINE_ARCH STREQUAL "arm64")
+            
+            # Homebrew dir is /opt/homebrew for arm64 (/usr/local for x86_64).
+            # Use METIS from Homebrew, which is currently 5.1.0 (IPOPT 3.12.8
+            # automatically pulls 4.10.0, which had some segfault issues)
+            set(HOMEBREW_DIR "/opt/homebrew")
+            set(METIS_INSTALL_DIR "${HOMEBREW_DIR}/opt/metis")
```

Comment:
Is homebrew the only way to get metis and are these directories hardcoded/unchangeable by users?

Reply:
> Is homebrew the only way to get metis

There are a few ways to get metis, but there seemed to be issues with using anything but the homebrew version:
1. Building from source from [here](http://glaros.dtc.umn.edu/gkhome/metis/metis/download): This built properly, however, the library that comes out does not seem to have the correct path stored in the .dylib when using `otool -L libmetis.dylib` to check dependencies (e.g., it seems relative rather than absolute), so there are linking problems downstream. I suspect there is some issue with the provided configure/cmake step and RPATH, as there is a warning about `MACOS_RPATH`.
2. Building from coin-or's repo `ThirdParty-Metis` ([link](https://github.com/coin-or-tools/ThirdParty-Metis)). The latest version here, however, is only 4.10.0, which had segfault issues downstream at runtime.
3. homebrew: automatically installs 5.1.0, and using this resolved segfaults.

> are these directories hardcoded/unchangeable by users

With homebrew, the directory structure is set by homebrew. For `arm64` installation, the base directory is `/opt/homebrew`. For `x86_64`, these are kept separately in `/usr/local`. From there, the relative directory of `opt/metis` should also be the same. These directories are set by homebrew when installing without any input from users. Whether users could change these afterwards, I'm not sure, but my understanding is that that would not be a typical use of homebrew.